src/graph/Node.cpp

```diff
@@ -180,3 +187,26 @@ cudaStream_t Node::getStreamHandle()
 {
 	return getGraphRunCtx()->getStream()->getHandle();
 }
+
+void Node::setPriority(int32_t requestedPriority)
+{
+	// It is illegal to set node's priority to a value lower than max priority value of its outputs.
+	if (!outputs.empty() && outputs.front()->priority > requestedPriority) {
+		auto msg = fmt::format("cannot set priority of node {} to {}, because it's children {} has higher priority {}",
+		                       getName(), requestedPriority, outputs.front()->getName(), outputs.front()->priority);
+		throw InvalidPipeline(msg);
+	}
+
+	this->priority = requestedPriority;
+	// Update parent nodes.
+	for (auto&& input : inputs) {
```

Comment:
I thought about this and it may be different from your intentions, but I will write this down, at least as a concept.

Although the above should work, maybe there is an option to implement simplier concept - where node prio is resolved only between outputs? In other words, node is always resolved before its children, and then we decide based on priority.

Reply:
Discussed offline and decided to go with the current implementation (global priorities).